src/rdkafka_idempotence.c

```diff
@@ -163,7 +163,7 @@ int rd_kafka_idemp_request_pid (rd_kafka_t *rk, rd_kafka_broker_t *rkb,
                 return 1;
         }
 
-        rd_kafka_wrunlock(rkb->rkb_rk);
+        rd_kafka_wrunlock(rk);
```

Comment:
do you also want to make this change in the if block above?

Reply:
good catch!